xrpl/testutil/integration/runner.go

```diff
@@ -0,0 +1,139 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/Peersyst/xrpl-go/pkg/crypto"
+	"github.com/Peersyst/xrpl-go/xrpl/queries/transactions"
+	"github.com/Peersyst/xrpl-go/xrpl/transaction"
+	"github.com/Peersyst/xrpl-go/xrpl/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+type Runner struct {
+	t      *testing.T
+	config *RunnerConfig
+
+	client  Client
+	wallets []*wallet.Wallet
+}
+
+// NewRunner creates a new runner. It doesn't connect to the websocket or generate wallets until Setup is called.
+// A testing.T is required to use the require package.
+func NewRunner(t *testing.T, client Client, config *RunnerConfig) *Runner {
+	return &Runner{
+		t:      t,
+		config: config,
+		client: client,
+	}
+}
+
+// Setup creates a new websocket client and generates the required number of wallets.
+// It also connects to the websocket and starts the client.
+// For every wallet, it will create a new account and fund it with the faucet.
+func (r *Runner) Setup() error {
+	if connectable, ok := r.client.(Connectable); ok {
+		err := connectable.Connect()
+		if err != nil {
+			return err
+		}
+	}
+
+	for i := 0; i < r.config.WalletCount; i++ {
+		w, err := wallet.New(crypto.ED25519())
+		if err != nil {
+			return err
+		}
+		err = r.client.FundWallet(&w)
+		if err != nil {
+			return err
+		}
+		r.wallets = append(r.wallets, &w)
+	}
+	return nil
+}
+
+// Teardown closes the websocket client.
+func (r *Runner) Teardown() error {
```

Comment:
Why not simply calling it `Disconnect` instead of `Teardown`?

Reply:
Because in the future it can contain more logic, for example, resetting a state. So it is not only disconnecting from the client